### ISSUE TITLE
Learning rate scaling with #GPUS

### ISSUE THREAD

**Issue by rakhimovv** [facebookresearch/detectron2#934]
The models in MODEL ZOO were trained using 8 GPUs.

Let's assume I have 16 GPUs or 4 GPUs and I keep the batch size the same as in the config. 

I know about the linear scaling rule, but that is about the connection between batch size and learning rate. What about #GPUS ~ base LR connection?

Should I scale base LR x0.5 in 1st case and x2 in 2nd case or just keep the same base lr?

My hypothesis that the answer is:
* no if one uses DataParallel model or single-GPU model and
* yes if DistributedDataParallel (like in this repo).

Am I right?

**Comment by ppwwyyxx** [comment]
It looks like you're asking a machine learning question - which we don't answer.

**Comment by ppwwyyxx** [comment]
If this is what you're asking: According to https://detectron2.readthedocs.io/modules/config.html#config-references, The (total) batch size is set by by `SOLVER.IMS_PER_BATCH`, regardless of the number of GPUS.

**Comment by rakhimovv** [comment]
@ppwwyyxx yeah, I understand that IMS_PER_BATCH is global batch size. the question is not ML question, it is Pytorch-related. Because is some repos I saw something like:
```
if distributed:
      lr /= world_size
```
but here I do not see modifications of BASE_LR in code. So it means that I should not change anything if I just change the number of gpus or should I scale lr properly then? (IMS_PER_BATCH is fixed)

**Comment by ppwwyyxx** [comment]
If the gradient is averaged across GPUs usually you don't need `      lr /= world_size`.
Still, this is not a question specific to detectron2, so it's best to ask in other places such as pytorch forum.
